datastruct/seqptrmap.c

```diff
@@ -56,8 +57,11 @@ seqptrmap_add(struct seqptrmap * M, void * ptr)
 		goto err0;
 	M->len += 1;
 
+	/* Check for overflow. */
+	assert(INT64_MAX - (int64_t)M->len >= M->offset);
```

Comment:
Need to assert that M->len <= (uint64_t)(INT64_MAX) first, since otherwise there could be a signed integer overflow when performing the cast.